packages/core/database/lib/connection.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable node/no-missing-require */
+/* eslint-disable node/no-extraneous-require */
+'use strict';
+
+const knex = require('knex');
+
+const SqliteClient = require('knex/lib/dialects/sqlite3/index');
+
+const trySqlitePackage = packageName => {
+  try {
+    require.resolve(packageName);
+    return packageName;
+  } catch (error) {
+    if (error.code === 'MODULE_NOT_FOUND') {
+      return false;
+    }
+    throw error;
+  }
+};
+class LegacySqliteClient extends SqliteClient {
+  _driver() {
+    return require('sqlite3');
+  }
+}
+
+const clientMap = {
+  'better-sqlite3': 'better-sqlite3',
+  '@vscode/sqlite3': 'sqlite',
+  sqlite3: LegacySqliteClient,
+};
+
+const getSqlitePackageName = () => {
+  // NOTE: allow forcing the package to use (mostly used for testing purposes)
+  if (typeof process.env.SQLITE_PKG !== undefined) {
+    return process.env.SQLITE_PKG;
```

Comment:
Is it supposed to be documented?

Reply:
IMO it doesn't makes sense to document it in the public doc but open to it

Reply:
Ok good. Me neither, just checking 👌

Reply:
We should still document it but can clarify it's mainly for testing.